src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -884,6 +887,55 @@ public async void StartNewReleaseAsync(string accountName, string projectName, A
                 body,
                 versionOverride: "5.0-preview.3",
                 baseAddressSubpath: "vsrm.");
+
+            return content.GetValue("id").ToObject<int>();
+        }
+
+        /// <summary>
+        /// Block until the release with ID informed finishes execution.
+        /// Checks the status of the release every 15 seconds.
+        /// </summary>
+        /// <param name="accountName"></param>
+        /// <param name="projectName"></param>
+        /// <param name="releaseId"></param>
+        public void WaitUntilCompleted(string accountName, string projectName, int releaseId)
+        {
+            while (true)
+            {
+                AzureDevOpsRelease rel = GetReleaseAsync(accountName, projectName, releaseId).GetAwaiter().GetResult();
```

Comment:
This is another deadlock problem.  Who is calling this that needs to spend a thread in this busy loop, and why aren't they async in the first place?

Reply:
The only caller of this is, itself, async. This whole method should be async, and they should wait for it... You have sync-over-async-over-sync-over-async.  This needs to get cleaned up so you stop switching contexts.